boot/bootstate20.go

```diff
@@ -38,6 +38,28 @@ func newBootState20(typ snap.Type) bootState {
 	}
 }
 
+//
+// modeenv methods
+//
+
+type bootState20Modeenv struct {
+	modeenv *Modeenv
+}
+
+func (bsm *bootState20Modeenv) loadModeenv() error {
+	// don't read modeenv multiple times
+	if bsm.modeenv != nil {
+		return nil
+	}
+	modeenv, err := ReadModeenv(dirs.GlobalRootDir)
```

Comment:
this can use "" to mean dirs.GlobalRootDir afair

Reply:
Yes, I changed this, though this exposed a small bug in one of the test helpers which I also fixed.